test/unit/ProgamCallUnit.js

```diff
@@ -22,21 +22,23 @@
 const { expect } = require('chai');
 const { ProgramCall } = require('../../lib/itoolkit');
 
+// now in this format
+// [name, type, value, options]
 const outBuf = [
-  [0, '10i0'],
-  [0, '10i0'],
-  ['', '36h'],
-  ['', '10A'],
-  ['', '1A'],
-  ['', '1A'],
-  [0, '10i0'],
-  [0, '10i0'],
+  ['', '10i0', 0],
+  ['', '10i0', 0],
+  ['', '36h', ''],
+  ['', '10A', ''],
+  ['', '1A', ''],
+  ['', '1A', ''],
+  ['', '10i0', 0],
+  ['', '10i0', 0],
```

Comment:
Don't these all need to be objects now?

Reply:
Just to confirm we are on the same page.

For this PR, How do we want DS to be defined?

Do we want DS to be an array of objects like shown here?
https://github.com/IBM/nodejs-itoolkit/issues/107#issuecomment-591632362

```js
const errno = [
  // fields can be defined using objects as well
  // attributes are set on the object itself, not in a separate object
  {name: 'bytes_provided', type:'10i0', setlen: 'rec2' },
  
  {name: 'bytes_available', type: '10i0'},

  // The value can be set on the object
  {name: 'msgid', type: '7a', value: 'XXX0000'},

  // we can also omit names in object format 
  {type: '1h'},

  // Data structures can also be nested using object format
  {name: 'message fields', type='ds', fields=[
      // You can always mix and match object/array fields
      ['field1', '10i0'],
      {name: 'field2', type:'10i0'}
  ]}
];
```

If that is the case then yes we will need to change these to objects.

Reply:
yes, all data types should be objects. We can add shortcuts in v1.1 or later.

Reply:
Sorry, that example was before we talked about the problem for ds options. Thus, ds that need options _must_ be objects, but I suggest we just stop supporting arrays for now as I just mentioned.

So something like this, instead:

```js
const errno = {
    name: "errno",
    type: "ds",
    setlen: "whatever",
    fields: [
        # fields here
    ]
}
```

Reply:
> but I suggest we just stop supporting arrays for now as I just mentioned.

Thanks! I agree ***all*** ds types will just be objects for now. We can add shortcuts and options later.

I will also need to update the `iPgm` wrapper to create `ds` types as objects.

Reply:
#157 will be affected as well.